src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,148 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASIC_LENGTH = 10;
+    private static final int LENGTH_STEP = 5;
+    private T[] list;
+    private int index;
+
+    public ArrayList() {
+        list = (T[]) new Object[BASIC_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (index == list.length) {
+            addSpase();
+        }
+        list[index] = value;
+        index++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index >= 0) {
+            if (index > this.index) {
+                /* If index of element doesn`t fall into the array */
+                throw new ArrayListIndexOutOfBoundsException("Incorrect index of list");
+            } else if (index == this.index) {
+                /* If index of element fall into the end of array */
+                add(value);
+            } else {
+                if (list.length == this.index) {
+                    /* If array is overflowing, we need increase the array */
+                    addSpase();
+                    addByIndex(value, index);
+                } else {
+                    addByIndex(value, index);
+                }
+            }
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is negative!");
```

Comment:
wrote a check above to avoid throwing an exception (if (index > this.index || index < 0))